src/mgsession.c

```diff
@@ -227,6 +228,10 @@ int mg_session_ensure_space_for_chunk(mg_session *session, size_t chunk_size) {
 
 int mg_session_read_chunk(mg_session *session) {
   uint16_t chunk_size;
+#ifdef __EMSCRIPTEN__
+  int socket = ((mg_raw_transport *)session->transport)->sockfd;
+  yield_until_async_read(socket, 10);
+#endif
```

Comment:
This is the second time, please make convert this to be used similarly to `mg_raw_transport_send`.